### DISCUSSION TITLE
How to register pinia‘s plugin in nuxt3?

### DISCUSSION THREAD

**Discussion by peterroe** [nuxt/nuxt#20459] [Questions]
I notice I can register vue plugin [this way](https://nuxt.com/docs/guide/directory-structure/plugins#vue-plugins)：
```ts
import VueGtag, { trackRouter } from 'vue-gtag-next'
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(VueGtag, {
    property: {
      id: 'GA_MEASUREMENT_ID'
    }
  })
  trackRouter(useRouter())
})
```

But how can i use pinia’s plugin in `nuxt3`?😊 https://pinia.vuejs.org/core-concepts/plugins.html

**Comment by SpiritusDeos** [comment]
1. Add pinia to your repo
2. Add this line In the `nuxt.config.ts` : `modules: [['@pinia/nuxt', { autoImports: ['defineStore'] }]],`
3. In the store directory create the different store that you need following this documentation https://pinia.vuejs.org/getting-started.html

**Comment by peterroe** [comment]
(*/ω＼*) Thank your reply. There is a problem with my formulation, I mean pinia's plugin, for example: https://pinia.vuejs.org/core-concepts/plugins.html
